source/MaterialXCore/Document.cpp

```diff
@@ -305,6 +305,30 @@ ValuePtr Document::getGeomPropValue(const string& geomPropName, const string& ge
     return value;
 }
 
+std::vector<OutputPtr> Document::getMaterialOutputs() const
+{
+    vector<OutputPtr> materialOutputs;
+
+    const std::string documentUri = getSourceUri();
```

Comment:
Within the MaterialX libraries, I would recommend using `string` instead of `std::string`, since it's more compact and gives us more flexibility in the future.